go/libraries/doltcore/doltdb/foreign_key_coll.go

```diff
@@ -134,6 +138,122 @@ func (fk ForeignKey) ValidateReferencedTableSchema(sch schema.Schema) error {
 	return nil
 }
 
+// HasDelayedResolution is true when a FK is created without a referenced table index and its relative tags.
+func (fk ForeignKey) HasDelayedResolution() bool {
+	return fk.ReferencedTableIndex == "" || fk.ReferencedTableColumns == nil
+}
+
+// ResolveReferencedIndexAndTags handles delayed foreign key resolution in the case that FOREIGN_KEY_CHECKS=0 when
+// a series of create table operations are done.
+func (fk ForeignKey) ResolveReferencedIndexAndTags(ctx context.Context, root *RootValue) (*RootValue, ForeignKey, error) {
```

Comment:
The function signature seems confused. If we are modifying the called foreign key, then this should be a pointer receiver. As well, since we are modifying the called key, we don't need to return it again. Either create and return a new foreign key, leaving the called one untouched, or remove the returned foreign key. Additionally, I don't think this should be modifying the root. The caller may not want to write the updated key to the root (or may want to delay it for validation, etc.) and root calls are expensive.

```golang
func (fk *ForeignKey) ResolveReferencedIndexAndTags(ctx context.Context, root *RootValue) error
```
or
```golang
func (fk ForeignKey) ResolveReferencedIndexAndTags(ctx context.Context, root *RootValue) (ForeignKey, error)
```